bin/kafka-server-stop.sh

```diff
@@ -36,7 +36,7 @@ else
     declare -a AbsolutePathToConfigArray
     for ((i = 0; i < ${#RelativePathArray[@]}; i++)); do
         AbsolutePathToConfig=$(readlink -f "${RelativePathArray[i]}")
-        if [ -z "$AbsolutePathToConfig" ]; then
+        if [ -z "$AbsolutePathToConfig" ] && [ -n "$INPUT_PROCESS_ROLE" ] || [ -n "$INPUT_NID" ]; then
```

Comment:
Is there a precedence issue here?

Reply:
You are right. Let me fix this.

Reply:
It might actually be best to rework things a bit to move the whole "AbsolutePathToConfigArray " section down until after line 53.

Reply:
yeah that will work. Let me change that.